### ISSUE TITLE
Support StatefulSets

### ISSUE THREAD

**Issue by isaacharrisholt** [DopplerHQ/kubernetes-operator#73]
From my testing, it doesn't seem like the operator has support for restarting pods defined in StatefulSets instead of Deployments. I may be doing it wrong, or there could be scope for improvement here - a couple of our microservices are defined this way rather than in deployments.

**Comment by watsonian** [comment]
@isaacharrisholt Right, we only support Deployments currently. Until we support other resource types, the way to go is using the [Reloader k8s controller](https://github.com/stakater/Reloader).

**Comment by isaacharrisholt** [comment]
@watsonian how much work would it be to add more resource types? It's something I'm potentially willing to look into.

**Comment by matimercado** [comment]
+1
